src/atom.rs

```diff
@@ -506,71 +506,89 @@ where
 
 ///////////////////////////////////////////////////////////////////////////////
 
-fn encode_primitive_dyn(col: &Arc<dyn Array>, row: usize) -> BytesText {
+fn encode_primitive_dyn(
+    col: &Arc<dyn Array>,
+    row: usize,
+) -> Result<BytesText, UnsupportedColumnType> {
+    let col_type = col.data_type().clone();
     if col.is_null(row) {
-        BytesText::new("null")
+        match col_type {
+            DataType::Int8
+            | DataType::Int16
+            | DataType::Int32
+            | DataType::Int64
+            | DataType::UInt8
+            | DataType::UInt16
+            | DataType::UInt32
+            | DataType::UInt64 => Ok(BytesText::new("0")),
+            DataType::Float16 | DataType::Float32 | DataType::Float64 => Ok(BytesText::new("0.0")),
+            // TODO add other types here
+            _ => Ok(BytesText::new("null")),
```

Comment:
This doesn't seem right to me -`null` and zero are two completely different values with very different meanings.

What is the motivation behind this?

The metadata endpoint should already be marking the nullable columns as `Nullable="true"` so the clients should treat them correctly.

Reply:
Using `null` value for the primitive types like Int16 will show the error from PowerBI when it try to read from the feed. 

But you are right, they are indeed different types, not sure why powerbi gave that error.

Reply:
Also tried to pass Nullable="true" but it doesn't seem to work though. will start debugging that instead.